lasagne/layers/autoenc.py

```diff
@@ -0,0 +1,262 @@
+                    b_idx += 1
+                else:
+                    kwargs = dict(b=b)
+
+                a_layer = BiasLayer(
+                    incoming=incoming,
+                    **kwargs
+                )
+            else:
+                a_layer = InverseLayer(
+                    incoming=incoming,
+                    layer=layer
+                )
+            autoencoder_layers.append(a_layer)
+
+    return autoencoder_layers[-1]
+
+
+def unfold_bias_and_nonlinearity_layers(layer):
+    """
+    Given a :class:`Layer` instance representing a stacked network,
+    this function adds a :class:`NonlinearityLayer` instance in between
+    each layer with attribute nonlinearity, with the same nonlinearity,
+    while setting the nonlinearity of the original layer to the identity
+    function.
+
+    This function is useful to build autoencoders with tied parameters.
+
+    Parameters
+    ----------
+        layer: :class:`Layer` instance
+            The :class:`Layer` instance with respect to wich the new
+            stacked Neural Network with added :class:`BiasLayer`: and
+            class:`NonlinearityLayer` are built.
+
+    Returns
+    -------
+        layer: :class:`Layer` instance
+           The output :class:`Layer` of the symmetric autoencoder with
+           tied weights.
+
+    Examples
+    --------
+    >>> import lasagne
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> from lasagne.layers import BiasLayer, NonlinearityLayer
+    >>> from lasagne.layers import unfold_bias_and_nonlinearity_layers
+    >>> from lasagne.layers import get_all_layers
+    >>> from lasagne.nonlinearities import tanh, sigmoid, identity
+    >>> l_in = InputLayer((100, 20))
+    >>> l1 = DenseLayer(l_in, num_units=50, nonlinearity=tanh)
+    >>> l_out = DenseLayer(l1, num_units=10, nonlinearity=sigmoid)
+    >>> l_out = unfold_bias_and_nonlinearity_layers(l_out)
+    >>> all_layers = get_all_layers(l_out)
+    >>> isinstance(all_layers[0], InputLayer)
+    True
+    >>> isinstance(all_layers[1], DenseLayer)
+    True
+    >>> all_layers[1].nonlinearity == identity
+    True
+    >>> all_layers[1].b is None
+    True
+    >>> isinstance(all_layers[2], BiasLayer)
+    True
+    >>> isinstance(all_layers[3], NonlinearityLayer)
+    True
+    >>> all_layers[3].nonlinearity == tanh
+    True
+    >>> isinstance(all_layers[4], DenseLayer)
+    True
+    >>> all_layers[4].nonlinearity == identity
+    True
+    >>> all_layers[4].b is None
+    True
+    >>> isinstance(all_layers[5], BiasLayer)
+    True
+    >>> isinstance(all_layers[6], NonlinearityLayer)
+    True
+    >>> all_layers[6].nonlinearity == sigmoid
+    True
+    """
+    layers = get_all_layers(layer)
+    incoming = layers[0]
+
+    for ii, layer in enumerate(layers[1:]):
+        layer.input_layer = incoming
+        if hasattr(layer, 'b') and not isinstance(layer, BiasLayer):
+            b = layer.b
+            layer.b = None
```

Comment:
That's not enough, you also need to remove it from their `params`:
```python
del layer.params[b]
```
It didn't hurt because `get_all_params` weeds out duplicates to deal with tied parameters already.